client/transport/tlsconfig_clone_go16.go

```diff
@@ -0,0 +1,13 @@
+// +build go1.6
+// +build !go1.7
+
+package transport
+
+import "crypto/tls"
+
+// TLSConfigClone returns a clone of tls.Config. This function is provided for
+// compatibility for go1.6 that doesn't include this method in stdlib.
+func TLSConfigClone(c *tls.Config) *tls.Config {
+	// returning a pointer to a local variable is legal in go
```

Comment:
Shouldn't we make a copy of stdlib `Config.clone()` function like there is for `go1.7` instead? I'm not sure how likely this is to cause actual problems for our use case but the problem here was that `tls.Config` contains mutexes that should never be copied.